### ISSUE TITLE
abort: http authorization

### ISSUE THREAD

**Issue by ozbillwang** [drone-plugins/drone-hg#16]
I can't wait the image `plugins/hg` to be available (#15), so I built the image and push to hub.docker.com directly.

But when I run the clone pipeline via `.drone.yml`, I got below error, which I don't have same  authorization issue with git. 

```
$ hg init .
$ hg pull --rev 1234d8904c030ce76a5d7e9dd32ef2c8d1e18467 https://bitbucket.org/org/project-1
pulling from https://bitbucket.org/org/project-1
abort: http authorization required for https://bitbucket.org/org/project-1
time="2017-08-25T10:56:51Z" level=fatal msg="exit status 255"
```

here is my `.drone.yml`

```
clone:
  hg:
    image: ozbillwang/hg
    path: bitbucket.org/org/project-1
```

Any suggestion?

**Comment by ozbillwang** [comment]
Read the code (https://github.com/drone-plugins/drone-hg/blob/3d24ff5e2f1cd9df1ac11a8442a881e320fa4190/main.go), seems this plugin is using different username/password. 

DRONE_NETRC_MACHINE
DRONE_NETRC_USERNAME
DRONE_NETRC_PASSWORD

So do I have to set again for `drone-hg` plugin? But how to set ?

**Comment by tboerger** [comment]
I think you can skip the path, and please try plugins/hg

**Comment by ozbillwang** [comment]
@tboerger 

Thanks to push the image`plugins/hg`. I did the test with it and get same error message.

```
clone:
  hg:
    image: plugins/hg
    path: bitbucket.org/org/project-1

```
We are trying to clone a mercurial repo from Bitbucket cloud. 
 
With git, Drone uses environments `DRONE_BITBUCKET_CLIENT` and `DRONE_BITBUCKET_SECRET` to get the repo read/write permission (which controlled by its bitbucket OAuth consumer) , but seems this plugin `plugins/hg` doesn't support these environments. 

So do I have to feed the `username/password` or apply SSH key to that repo to avoid `http authorization required` issue?

When run `env` in plugin `plugins/hg` or other images, I don't see these environments. So I am curious how git plugins (`plugins/git`) works with them?

**Comment by tboerger** [comment]
Than somebody should compare the git plugin env vars with these vars. And somebody should test the plugin standalone.

**Comment by tboerger** [comment]
Solved via #17
